### PR TITLE
Add ability to do multi-branch if / else logic

### PR DESCRIPTION
**Pull request by jeffmay**

### CODE REVIEW

**Review by paul-a-kennedy-rally — APPROVED** [review]
Looks good--nice tests.

File: core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/WhenExprSpec.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          .elseReturn(const(c))
+        val res = eval(FactTable.empty)(q)
+        assertResult(c)(res.output.value)
+      }
+
+      "return the third branch of a if / elif / elif / else when the first two conditions are 'false' and the third condition is 'true'" in {
+        val a = "A"
+        val b = "B"
+        val c = "C"
+        val d = "D"
```
Comment (jeffmay) [inline core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/WhenExprSpec.scala]: I might as well move these to the top...

File: core/src/main/scala/com/rallyhealth/vapors/factfilter/dsl/ExprDsl.scala [hunk 2]
```diff
@@ -125,22 +125,31 @@ object ExprDsl extends ExprBuilderSyntax with ExprBuilderCatsInstances {
...
-  final class WhenThenBuilder[F[_], V, R, P](
-    condExpr: CondExpr[F, V, P],
-    thenExpr: Expr[F, V, R, P],
-  ) {
+  final class ElifBuilder[F[_], V, R, P](
+    private val t: (CondExpr[F, V, P], NonEmptyList[ConditionBranch[F, V, R, P]]),
+  ) extends AnyVal {
+
+    def thenReturn(thenExpr: Expr[F, V, R, P]): ElseDefaultBuilder[F, V, R, P] =
+      new ElseDefaultBuilder(t._2 ::: NonEmptyList.of(ConditionBranch(t._1, thenExpr)))
```
Comment (jeffmay) [inline core/src/main/scala/com/rallyhealth/vapors/factfilter/dsl/ExprDsl.scala]: Maybe I should use a `Queue` to be more efficient in building these.

Comment (jeffmay) [reply core/src/main/scala/com/rallyhealth/vapors/factfilter/dsl/ExprDsl.scala]: Actually, I'll just reverse the list at the end.

**Review by darmbrus — APPROVED** [review]

**Pull request merged**